includes/WikiPage.php

```diff
@@ -590,9 +590,11 @@ public function isCountable( $editInfo = false ) {
 	 *
 	 * The target will be fetched from the redirect table if possible.
 	 * If this page doesn't have an entry there, call insertRedirect()
+	 *
+	 * @param bool $userMaster
 	 * @return Title|mixed object, or null if this page is not a redirect
 	 */
-	public function getRedirectTarget() {
+	public function getRedirectTarget( $userMaster = false ) {
 		if ( !$this->mTitle->isRedirect() ) {
```

Comment:
The ```isRedirect``` method does a DB lookup as well to determine whether the page is a redirect, and it accepts the ```\Title::GAID_FOR_UPDATE``` argument.